src/js/media-display-gesture-receiver.js

```diff
@@ -0,0 +1,142 @@
+import { MediaUIAttributes, MediaUIEvents, PointerTypes } from './constants.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  :host {
+    display: inline-block;
+    width: auto;
+    height: auto;
+    box-sizing: border-box;
+
+    pointer-events: auto;
+  }
+</style>
+`;
+
+class MediaDisplayGestureReceiver extends window.HTMLElement {
+  // NOTE: Currently "baking in" actions + attrs until we come up with
+  // a more robust architecture (CJP)
+  static get observedAttributes() {
+    return [MediaUIAttributes.MEDIA_CONTROLLER, MediaUIAttributes.MEDIA_PAUSED];
+  }
+
+  constructor(options = {}) {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const buttonHTML = template.content.cloneNode(true);
+    this.nativeEl = buttonHTML;
+
+    // Slots
+    let slotTemplate = options.slotTemplate;
+
+    if (!slotTemplate) {
+      slotTemplate = document.createElement('template');
+      slotTemplate.innerHTML = `<slot>${options.defaultContent || ''}</slot>`;
+    }
+
+    this.nativeEl.appendChild(slotTemplate.content.cloneNode(true));
+
+    shadow.appendChild(buttonHTML);
+
+    this._pointerType = undefined;
+    const pointerDownHandler = (e) => {
+      // Since not all browsers have updated to be spec compliant, where 'click' events should be PointerEvents,
+      // we can use use 'pointerdown' to reliably determine the pointer type. (CJP).
+      this._pointerType = e.pointerType;
+    };
+
+    this.addEventListener('pointerdown', pointerDownHandler);
+
+    this.addEventListener('click', (event) => {
+      if (!event.pointerType) {
+        event.pointerType = this._pointerType;
+      }
+      const { pointerType } = event;
```

Comment:
nit: we can set a default value via destructuring so that we don't modify the native event object.
```suggestion
      const { pointerType = this._pointerType } = event;
```

Reply:
I was debating that. I thought it would be good to pass the "expected" pointerType to the handler. Thoughts?